### PR TITLE
Create external table fdw extension under gpcontrib.

### PR DESCRIPTION
**Pull request by JunfengYang**
Remove pg_exttable.h since the catalog is no longer exist anymore.
Move function declaration in pg_exttable.h into external.h.
Extract related code into external.c which maintains all codes that
can not be moved into an external table fdw extension.

Also move the external table orca interface into external.c as a workaround.
Maybe provide orca fdw routine in the future.

Extract the external table's execution logic into external table fdw
extension.

Create the gp_exttable_fdw extension during gpinitsystem to allow
creating system external tables.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/catalog/gp_toolkit.sql [hunk 1]
```diff
@@ -180,6 +180,8 @@ AS
 GRANT SELECT ON TABLE gp_toolkit.__gp_number_of_segments TO public;
 
 
+CREATE EXTENSION gp_exttable_fdw;
```
Comment (adam8157) [inline src/backend/catalog/gp_toolkit.sql:183]: Would gp_exttable_fdw installing another sql file into cdb_init.d be better?

**Comment by adam8157** [comment]
Looks good so far.

```
adam=# drop extension gp_exttable_fdw cascade;
...
adam=# create external table test (c1 int) location('gpfdist://xx/xx') format 'csv';
ERROR:  server "gp_exttable_server" does not exist
```

File: src/backend/catalog/gp_toolkit.sql (hunk 1 above)
Comment (JunfengYang) [reply src/backend/catalog/gp_toolkit.sql:183]: What are the benefits of doing this?

**Comment by hlinnaka** [comment]
I have no objections to this, although this doesn't make much difference, just moves code around. I would be more excited about removing all the code from the backend that currently calls `rel_is_external_table()`.

**Comment by JunfengYang** [comment]
@hlinnaka, we want to deliver small PR to let the PR easily understand. Also, the FDW may affect lot's of teams, so we want to discuss first before we change anything.

`rel_is_external_table` this function is used a lot, we plan to use FDW native routines to replace them one by one.
For example, in the coming FDW enhancement proposal, we plan to fully support foreign table data distribution(Currently we hack the GpPolicy for foreign table and even external table. Replicated and Partitioned distribution is not supported).
This will remove the `rel_is_external_table` call in `GpPolicyFetch`.

**Pull request merged**